lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -88,4 +100,25 @@ def valid(opts)
     return true
   end
 
+  def use_id_or_get_id_from_host_address(opts, method)
+    begin
+      if opts[:id]
+        self.data_service_operation do |data_service|
+          data_service.send(method, opts)
+        end
+      elsif opts[:address]
+        self.data_service_operation do |data_service|
+          host = data_service.get_host(opts)
+
+          if host
+            opts[:id] = host.id
+            data_service.send(method, opts.reject {|k,v| k==:address})
```

Comment:
I don't understand why you're sending through the options hash without the address? why do you need to remove it

Reply:
The tag methods at the DB layer only use the ID of a Host to perform operations, the address isn't used so it doesn't need to be sent

Reply:
ok...but is there a need to specifically remove `address` strictly speaking couldn't any number of values be being passed in opts that you _aren't_ removing?

Reply:
True, in the above changes I know that address isn't needed but there could be 1000 items in opts for all I know. I'll pull out the reject 👍